docs/testsuite_overview.md

```diff
@@ -0,0 +1,255 @@
+  Tests cover schema types (records, unions, enums, typerefs, ...), primitive types, optionals and defaults.
+  Backward compatibility rules are also covered.
+  
+* **Wire protocol: Building requests and decoding responses**  
+  Tests cover serializing/deserializing of URLs, 
+  Headers and bodies, Escaping/encoding, batch formats, projections, and partial updates.
+
+### JSON Tests
+These are tests for serialization and deserialization. 
+
+| JSON feature | Details |
+|--------------|---------|
+| basic types| string, number, boolean, null, object, array|
+| Empty collections| empty object, empty array, object with empty arrays, object with empty objects, array of empty object, array of empty arrays| 
+| large numbers |i32 and i64|
+| special characters| periods in key|
+| unicode| Chinese character and e with an accent mark|
+
+
+### Data Schema Tests
+These are tests for data template generation from schema. 
+
+| Schema feature | Details|
+|------------------|--------|
+| Primitive types|int, long, float, double, bytes, string |
+| Complex types | array of maps, map of ints, record with props |
+| Complex type: Unions | union of complex types, union of primitives, union of same types |
+| Enums | With props and with alias|
+| Fixed type | |
+| Typerefs | for string, array, chained typeref, array, map, field, union|
+| Include | include, and include with include |
+| Default Fixup | to see if required fields are "fixed up" with defaults|
+| Optional fields | |
+
+### Wire Protocol Tests
+These are tests for building requests and decoding responses. 
+
+We test for well-formed requests by comparing the built http request
+with the expected http request in the requests/ or requests-v2/ folder. We compare url, method, headers, and body.
+We test that Rest.li can decode a Rest.li response from an http response by checking the decoded Rest.li response for
+the correct values. We compare the response's status and error message with the status and error message specified by 
+manifest.json. The body of the response is tested through manual assertions that check for the correct values.
+
+##### Basic Resource Method Tests for Requests/Responses
+
+| Rest.li Method | Collection | Simple | Association | Action Set|
+|--------|------|----------------|----------------|------------------|
+| get | x | x | x | |
+| batch-get | x |  | x | |
+| finder | x | | x | |
+| create | x | | | |
+| create with returned entity | x | | | |
+| batch-create | x | | | |
+| update | x | x | x | |
+| partial update| x | |o| |
+| batch-update| x |  | x | |
+| batch-partial-update| x| | o | |
+| delete | x | x | x | | 
+| batch-delete | x | | x | | 
+| action | o |  o | o | x | 
+
+"x" - test is included in Rest.li Test Suite Specification
+"o" - test is not included but method should be supported by the resource 
+" " - test is not included and method should NOT be supported by resource
+
+##### Resource Key Tests
```

Comment:
Leave some space between the header and table below it, right now, it is too crowded in my preview.